js/main.js

```diff
@@ -157,14 +156,51 @@ function switchView(string) {
   }
 }
 
-$entries.addEventListener('click', handleSwap);
-$newBtn.addEventListener('click', handleSwap);
 
 function handleSwap(event) {
   var entryAttr = event.target.getAttribute('data-view');
   data.view = entryAttr;
   switchView(entryAttr);
   if (entryAttr === 'entry-form') {
     $entryTypeHeading.textContent = 'New Entry';
+    $deleteLink.classList.add('visiblity-hidden');
+  }
+}
+
+var modalOpen = false;
+function handleDeleteModal(event) {
+  if (modalOpen === false) {
+    $modalContainer.classList.remove('hidden');
+    modalOpen = true;
+  }
+}
+
+function handleHideModal(event) {
+  if (modalOpen === true) {
+    $modalContainer.classList.add('hidden');
+    modalOpen = false;
   }
 }
+
+function handleConfirmModal(event) {
+  for (var i = 0; i < data.entries.length; i++) {
+    if (data.editing.entryId === data.entries[i].entryId) {
+      data.entries.splice(i, 1);
+      var $entryToDelete = document.querySelector('[data-entry-id]');
```

Comment:
Wouldn't this just delete the first entry in the list? What happens if you delete the second entry or one later in the list?